### PR TITLE
[1] Cauchy

### PR DESCRIPTION
**Pull request by jonpsy**
I can optimize the Cauchy Kernel much further, for instance, using gamma auto parameter instead of sigma, but I beleive we can handle it on seperate PR. I've done the minimal work so that it derives from SIV Kernel. #5151.

### CODE REVIEW

File: src/shogun/kernel/CauchyKernel.cpp [hunk 1]
```diff
@@ -49,8 +52,10 @@ bool CauchyKernel::init(std::shared_ptr<Features> l, std::shared_ptr<Features> r
 
 void CauchyKernel::init()
 {
+	auto dist = std::make_shared<EuclideanDistance>();
+	m_distance = dist;
```
Comment (gf712) [inline src/shogun/kernel/CauchyKernel.cpp]:
```suggestion
	m_distance = std::make_shared<EuclideanDistance>();
```
no?

File: src/shogun/kernel/CauchyKernel.cpp [hunk 2]
```diff
@@ -49,8 +52,10 @@ bool CauchyKernel::init(std::shared_ptr<Features> l, std::shared_ptr<Features> r
...
+
 	SG_ADD(&m_sigma, "sigma", "Sigma kernel parameter.", ParameterProperties::HYPER);
-	SG_ADD(&m_distance, "distance", "Distance to be used.", ParameterProperties::HYPER);
```
Comment (gf712) [inline src/shogun/kernel/CauchyKernel.cpp:53]: is this registered by shift invariant?

File: examples/meta/src/kernel/cauchy.sg.in [hunk 3]
```diff
@@ -7,9 +7,7 @@ Features feats_test = create_features(f_feats_test)
 #![create_features]
 
 #![create_kernel]
-Distance dist = create_distance("EuclideanDistance")
-dist.init(feats_train, feats_train)
-Kernel cauchy = create_kernel("CauchyKernel", sigma=1.0, distance=dist)
+Kernel cauchy = create_kernel("CauchyKernel", sigma=1.0)
```
Comment (gf712) [inline examples/meta/src/kernel/cauchy.sg.in:10]: you can still pass along the distance object explicitly no?

File: src/shogun/kernel/CauchyKernel.cpp (hunk 2 above)
Comment (jonpsy) [reply src/shogun/kernel/CauchyKernel.cpp:53]: Yes :)

File: examples/meta/src/kernel/cauchy.sg.in (hunk 3 above)
Comment (jonpsy) [reply examples/meta/src/kernel/cauchy.sg.in:10]: Of course, but what's to stop the users to using a distance which isn't aligned with the formula? 

For ex: Circular uses L1 as per formula but users can use L2 too.

 Wouldn't that lead to confusion?

**Comment by jonpsy** [comment]
Re-open when devs are back.

**Pull request closed without merging**